integreat_cms/google_translate_api/google_translate_api_client.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import annotations
+
+import logging
+from html import unescape
+from typing import TYPE_CHECKING
+
+from django.apps import apps
+from django.conf import settings
+from django.contrib import messages
+from django.db import transaction
+from django.utils.translation import gettext_lazy as _
+from google.auth.exceptions import DefaultCredentialsError
+from google.cloud import translate_v2, translate_v3  # type: ignore[attr-defined]
+from google.oauth2 import service_account
+
+from ..core.utils.machine_translation_api_client import MachineTranslationApiClient
+from ..core.utils.machine_translation_provider import MachineTranslationProvider
+from ..textlab_api.utils import check_hix_score
+
+if TYPE_CHECKING:
+    from django.forms.models import ModelFormMetaclass
+    from django.http import HttpRequest
+
+    from integreat_cms.cms.models.events.event import Event
+    from integreat_cms.cms.models.events.event_translation import EventTranslation
+    from integreat_cms.cms.models.languages.language import Language
+    from integreat_cms.cms.models.pages.page import Page
+    from integreat_cms.cms.models.pages.page_translation import PageTranslation
+    from integreat_cms.cms.models.pois.poi import POI
+    from integreat_cms.cms.models.pois.poi_translation import POITranslation
+    from integreat_cms.cms.models.regions.region import Region
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleTranslateApiClient(MachineTranslationApiClient):
+    """
+    DeepL API client to automatically translate selected objects.
+    """
+
+    def __init__(self, request: HttpRequest, form_class: ModelFormMetaclass) -> None:
+        """
+        Initialize the Google Translate client
+
+        :param region: The current region
+        :param form_class: The :class:`~integreat_cms.cms.forms.custom_content_model_form.CustomContentModelForm`
+                           subclass of the current content type
+        """
+        super().__init__(request, form_class)
+        if not MachineTranslationProvider.is_permitted(
+            request.region, request.user, form_class._meta.model
+        ):
+            raise RuntimeError(
+                f'Machine translations are disabled for content type "{form_class._meta.model}" and {request.user!r}.'
+            )
+        if not settings.GOOGLE_TRANSLATE_ENABLED:
+            raise RuntimeError("Google translate is disabled globally.")
+
+        try:
+            credentials = service_account.Credentials.from_service_account_file(
+                settings.GOOGLE_APPLICATION_CREDENTIALS
+            )
+        except DefaultCredentialsError as e:
```

Comment:
Do you know how we can get this error here?
Maybe I misunderstood that, but it seems to me, that this error can occur, if we use default credentials - taken from an environment variable. So it's not our case.